src/cloudant/_common_util.py

```diff
@@ -287,66 +290,187 @@ def __new__(cls, code):
             return str.__new__(cls, code.encode('utf8'))
         return str.__new__(cls, code)
 
-class InfiniteSession(Session):
+
+class ClientSession(Session):
```

Comment:
The majority of the work that `CookieSession` and `IAMSession` have to do is the same, other than the extra piece to get an IAM token (and the content-type of the `POST` to `*_session` which should be `application/json` for IAM).
It would be nice (but by no means essential if it is harder than it initially looks) if we could use the same login/logout/expired checks in this base class and just customize them slightly (e.g. with the url/data/response conditions).